swift/apple/FirezoneKit/Sources/FirezoneKit/Views/ResourceMenuSection.swift

```diff
@@ -0,0 +1,277 @@
+//
+//  ResourceMenuSection.swift
+//  (c) 2025 Firezone, Inc.
+//  LICENSE: Apache-2.0
+//
+
+import SwiftUI
+
+/// Individual resource menu item with submenu containing resource details
+struct ResourceMenuItem: View {
+  let resource: Resource
+  @EnvironmentObject var store: Store
+
+  var body: some View {
+    Menu {
+      ResourceDetailsSubmenu(resource: resource)
+    } label: {
+      HStack(spacing: 4) {
+        if let symbolName = resourceSymbol {
+          Image(systemName: symbolName)
+        }
+        Text(resource.name)
+      }
+    }
+  }
+
+  var resourceSymbol: String? {
+    if resource.isInternetResource() {
+      return store.configuration.internetResourceEnabled ? "network" : "network.slash"
+    }
+    // For regular resources, show status indicator
```

Comment:
Similar to the comment above: if we're changing how the resource list appears on macOS, do we need to update other platforms as well?

In addition, we may want to think about other states the resource might be in in the future, such as access denied and peer to peer vs relayed, and thus be careful about which icons we pick here to avoid ambiguity or clashing with a future resource state.

See https://github.com/firezone/firezone/issues/6158 for example.

Considering the above, wdyt of using simply `circle.fill` for the connected state and plain `circle` for unknown or "offline" state? We can't detect that a resource is offline, rather, this signifies all gateways for the resource are offline.